frontend/catalyst/compiler.py

```diff
@@ -75,89 +77,197 @@ def get_executable_path(project, tool):
-    "--canonicalize",
-    "--chlo-legalize-to-hlo",
-    "--mhlo-legalize-control-flow",
-    "--hlo-legalize-to-linalg",
-    "--mhlo-legalize-to-std",
-    "--convert-to-signless",
-    "--canonicalize",
-]
-
-quantum_compilation_pass_pipeline = [
-    "--lower-gradients",
-]
-
-bufferization_pass_pipeline = [
-    "--inline",
-    "--gradient-bufferize",
-    "--scf-bufferize",
-    "--convert-tensor-to-linalg",  # tensor.pad
-    "--convert-elementwise-to-linalg",  # Must be run before --arith-bufferize
-    "--arith-bufferize",
-    "--empty-tensor-to-alloc-tensor",
-    "--bufferization-bufferize",
-    "--tensor-bufferize",
-    "--linalg-bufferize",
-    "--tensor-bufferize",
-    "--quantum-bufferize",
-    "--func-bufferize",
-    "--finalizing-bufferize",
-    "--buffer-hoisting",
-    "--buffer-loop-hoisting",
-    "--promote-buffers-to-stack",
-    "--buffer-deallocation",
-    "--convert-bufferization-to-memref",
-    "--canonicalize",
-    "--cse",
-]
-
-llvm_lowering_pass_pipeline = [
-    "--convert-linalg-to-loops",
-    "--convert-scf-to-cf",
-    # This pass expands memref operations that modify the metadata of a memref (sizes, offsets,
-    # stdies) into a sequence of easier to analyze constructs. In particular, this pass transforms
-    # operations into explicit sequence of operations that model the effect of this operation on the
-    # different metadata. This pass uses affine constructs to materialize these effects.
-    # Concretely, expanded-strided-metadata is used to decompose memref.subview as it has no
-    # lowering in -convert-memref-to-llvm.
-    "--expand-strided-metadata",
-    "--lower-affine",
-    "--convert-complex-to-standard",  # added for complex.exp lowering
-    "--convert-complex-to-llvm",
-    "--convert-math-to-llvm",
-    # Must be run after -convert-math-to-llvm as it marks math::powf illegal but doesn't convert it.
-    "--convert-math-to-libm",
-    "--convert-arith-to-llvm",
-    "--convert-memref-to-llvm=use-generic-functions",
-    "--convert-index-to-llvm",
-    "--convert-gradient-to-llvm",
-    "--convert-quantum-to-llvm",
-    # Remove any dead casts as the final pass expects to remove all existing casts,
-    # but only those that form a loop back to the original type.
-    "--canonicalize",
-    "--reconcile-unrealized-casts",
-]
-
-compiler = get_executable_path("llvm", "llc")
-compiler_flags = [
-    "--filetype=obj",
-    "--relocation-model=pic",
-]
+    if INSTALLED:
+        return os.path.join(package_root, "lib")  # pragma: no cover
+    return os.getenv(env_var, default_lib_paths.get(project, ""))
+
+
+class PassPipeline(abc.ABC):
+    """Abstract PassPipeline class."""
+
+    _executable: Optional[str] = None
+    _default_flags: Optional[list[str]] = None
+
+    @staticmethod
+    @abc.abstractmethod
+    def get_output_filename(infile):
+        """Compute the output filename from the input filename.
+
+        Derived classes are expected to implement this method.
+
```

Comment:
You may want to consider this suggestion,
```suggestion
        .. note:

                Derived classes are expected to implement this method.

```